ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -876,3 +881,31 @@ func (ipfs *Connector) SwarmPeers() (api.SwarmPeers, error) {
 	}
 	return swarm, nil
 }
+
+// BlockPut triggers an ipfs block put on the given data, inserting the block
+// into the ipfs daemon's repo.
+func (ipfs *Connector) BlockPut(data []byte) (api.Pin, error) {
```

Comment:
This is abusing a little bit the `api.Pin` type to serialize a Cid. Perhaps we'll be fine just returning a string (unless you think we might need to return more information and thus need a wrapper type in the future).